contrib/opencensus-ext-azure/opencensus/ext/azure/common/transport.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import requests
+
+from opencensus.trace import execution_context
+
+logger = logging.getLogger(__name__)
+
+
+class TransportMixin(object):
+    def _transmit_from_storage(self):
+        for blob in self.storage.gets():
+            if blob.lease(self.options.timeout + 5):
+                envelopes = blob.get()  # TODO: handle error
+                result = self._transmit(envelopes)
+                if result > 0:
+                    blob.lease(result)
+                else:
+                    blob.delete(silent=True)
+
+    def _transmit(self, envelopes):
+        """
+        Transmit the data envelopes to the ingestion service.
+        Return a negative value for partial success or non-retryable failure.
+        Return 0 if all envelopes have been successfully ingested.
+        Return the next retry time in seconds for retryable failure.
+        This function should never throw exception.
+        """
+        # TODO: prevent requests being tracked
+        blacklist_hostnames = execution_context.get_opencensus_attr(
```

Comment:
Not sure if it's a good idea to put this in the trace `execution_context`? /cc @c24t WDYT?

Reply:
Long term goal is to avoid exporters to access execution_context directly, for more info, refer to #573.
In this PR, the code was extracted from the existing trace exporter.

Reply:
I'll try to address it in a separate PR, there are similar scenarios which we need a consistent solution:
1. How to prevent traces generated by trace exporters (e.g. during sending data via HTTP) being sent to trace exporter and cause infinite loop.
2. How to prevent logs/exceptions generated by log exporters/handlers being emitted to the same log handlers and cause infinite loop.

My current thinking is to have the entire worker thread marked as "do not track", so all the spans/logs will have a special tag, and based on this tag we can filter these data out before sending them to exporters.